test/cppsuite/test_harness/configuration_settings.h

```diff
@@ -1,27 +1,32 @@
-/* Include guard. */
 #ifndef CONFIGURATION_SETTINGS_H
 #define CONFIGURATION_SETTINGS_H
 
-#include <string>
 #include <stdexcept>
+#include <string>
+
+extern "C" {
+#include "test_util.h"
+}
 
 #include "wt_internal.h"
 
 namespace test_harness {
 class configuration {
     public:
-    configuration(const char *test_config_name, const char *config) : _config(config)
+    configuration(const std::string &test_config_name, const std::string &config) : _config(config)
     {
-        int ret = wiredtiger_config_parser_open(nullptr, config, strlen(config), &_config_parser);
+        int ret =
+          wiredtiger_config_parser_open(nullptr, config.c_str(), config.size(), &_config_parser);
         if (ret != 0)
             throw std::invalid_argument(
               "failed to create configuration parser for provided config");
-        if (wiredtiger_test_config_validate(nullptr, nullptr, test_config_name, config) != 0)
+        if (wiredtiger_test_config_validate(
```

Comment:
We should validate prior to creating the configuration parser, I know that this wasn't your code originally but could you fix that here?